sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -229,33 +229,58 @@ case class RLike(left: Expression, right: Expression) extends StringRegexExpress
 
 
 /**
- * Splits str around pat (pattern is a regular expression).
+ * Splits str around matches of the given regex.
  */
 @ExpressionDescription(
-  usage = "_FUNC_(str, regex) - Splits `str` around occurrences that match `regex`.",
+  usage = "_FUNC_(str, regex, limit) - Splits `str` around occurrences that match `regex`" +
+    " and returns an array of at most `limit`",
+  arguments = """
+    Arguments:
+      * str - a string expression to split.
+      * regex - a string representing a regular expression. The regex string should be a
+        Java regular expression.
+      * limit - an integer expression which controls the number of times the regex is applied.
+
+        limit > 0: The resulting array's length will not be more than `limit`, and the resulting
+                   array's last entry will contain all input beyond the last matched regex.
+
+        limit < 0: `regex` will be applied as many times as possible, and the resulting
+                   array can be of any size.
+
+        limit = 0: `regex` will be applied as many times as possible, the resulting array can
```

Comment:
why do we want to build into split to remove trailing empty strings?

Reply:
@rxin this is default behavior for java lang string's `split` implementation. https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int), nothing has been specifically built into it here.

Reply:
yea but i'd focus on what behavior we want to enable. do other database systems have this split=0 semantics? if not, i'd rewrite split=0 internally to just -1.

Reply:
I see this as a value-add in that the user just gets more ammunition to answer a given question about his/her data. Sure, it's not necessary to have the `limit = 0` case, but if a user can get what they need by writing fewer lines of code because it exists I'd say it's definitely worth exposing.